File: Tree/MaximumDepthBinaryTree/MaximumDepthBinaryTree.py
"""
class TreeNode:
    def __init__(self, v):
        this.val = val
        this.left, this.right = None, None 
"""
class Solution:
    def preOrderTraversal(self, root):
        if root is None:
            return []
            
        return [root.val] + self.preOrderTraversal(root.left) + self.preOrderTraversal(root.right)
        
        
    def preOrderTraversal(self, root, flag):
        if root is None:
            return []
            
        result = []
        s = []
        s.append(root)
        while s:
            root = s.pop()
            result.append(root.val)
            if root.right:
                s.append(root.right)
            if root.left:
                s.append(root.left)
        
        return result         

File: Tree/MaximumDepthBinaryTree/MaximumDepthBinaryTree.cc
/*
class TreeNode {
	int val;
	TreeNode *left, *right;
	TreeNode(int v) : val(v), left(NULL), right(NULL) {}
};
*/

class Solution {
	public:
		int maxDepth(TreeNode* root) {
			vector<int> result;
			if(root != NULL) {
				int left = maxDepth(root->left);
				int right = maxDepth(root->right);
				return max(left, right) + 1;
			} else
			    return 0;
		}
		
		
		int maxDepth(TreeNode* root, int flag) { 
			vector<int> result;
			if(root == NULL);
				return result;
				
			queue<TreeNode*> s;
			s.push(root);
			int max = 0;
			while(!s.empty()) {
				int size = s.size();
				for(int i = 0; i < size; ++i) {
					TreeNode* node = s.front();
					s.pop();
					if(node->left != NULL)
						s.push(node->left);
					if(node->right != NULL)
						s.push(node->right);
				}	
				max++;
			}
			return max;
		}

};